src/Particle/ParticleSet.cpp

```diff
@@ -402,6 +402,14 @@ void ParticleSet::makeMove(Index_t iat, const SingleParticlePos_t& displ)
   computeNewPosDistTablesAndSK(iat, activePos);
 }
 
+void ParticleSet::makeMove(Index_t iat, const SingleParticlePos_t& displ, const RealType& sdispl)
```

Comment:
Rename makeMoveWithSpin.
Please maximize code reuse. Instead of copy code.
{
  makeMove();
  activeSpin = spins[iat] + sdispl;
}
Please do similar change in makeMoveAndCheck.